crates/wasmprinter/src/operator.rs

```diff
@@ -424,10 +440,37 @@ macro_rules! define_visit {
             )?;
         }
     );
-    (payload $self:ident StructNewDefault $type_index: ident) => (
+    (payload $self:ident StructNewDefault $type_index:ident) => (
+        $self.type_index_no_type($type_index)?;
+    );
+    (payload $self:ident ArrayNew $type_index:ident) => (
+        $self.type_index_no_type($type_index)?;
+    );
+    (payload $self:ident ArrayNewDefault $type_index:ident) => (
+        $self.type_index_no_type($type_index)?;
+    );
+    (payload $self:ident ArrayNewFixed $type_index:ident $n:ident) => (
+        $self.type_index_no_type($type_index)?;
+        $self.print_u32($n);
+    );
+    (payload $self:ident ArrayNewData $type_index:ident $data_index:ident) => (
+        $self.type_index_no_type($type_index)?;
+        $self.data_index_no_data($data_index)?;
+    );
+    (payload $self:ident ArrayNewElem $type_index:ident $elem_index:ident) => (
+        $self.type_index_no_type($type_index)?;
+        $self.elem_index_no_elem($elem_index)?;
+    );
+    (payload $self:ident ArrayGet $type_index:ident) => (
+        $self.type_index_no_type($type_index)?;
+    );
+    (payload $self:ident ArrayGetS $type_index:ident) => (
+        $self.type_index_no_type($type_index)?;
+    );
```

Comment:
To avoid needing to list all the instructions manually could these all have a slightly different name for their field? Maybe something like `array_type_index` or something like that? Either way works but the intention of these macros was to avoid case-per-instruction variants where possible.